src/http_interop.rs

```diff
@@ -188,26 +188,31 @@ impl From<Response> for http::Response<Vec<u8>> {
 /// ```
 impl From<http::request::Builder> for Request {
     fn from(value: http::request::Builder) -> Self {
-        let mut new_request = crate::agent().request(
-            value.method_ref().map_or("GET", |m| m.as_str()),
-            &value
-                .uri_ref()
-                .map_or("https://example.com".to_string(), |u| u.to_string()),
-        );
+        convert_from_http_builder_with_agent(value, crate::agent())
+    }
+}
 
-        if let Some(headers) = value.headers_ref() {
-            for (name, value) in headers {
-                let mut raw_header: Vec<u8> = name.to_string().into_bytes();
-                raw_header.extend(b": ");
-                raw_header.extend(value.as_bytes());
-                let header = HeaderLine::from(raw_header).into_header().unwrap();
+/// Converts an [`http::request::Builder`] into a [`Request`] with a custom [`Agent`].
+pub fn convert_from_http_builder_with_agent(value: http::request::Builder, agent: crate::Agent) -> Request {
```

Comment:
`agent` is used only for the `&self` `request` call, so maybe take an `&crate::Agent` here?